File: kachelib/src/main/java/com/mskn73/kache/Serializer.java
package com.mskn73.kache;

import com.google.gson.Gson;

/**
 * Json Serializer/Deserializer.
 */
public class Serializer {

  private final Gson gson = new Gson();

  Serializer() {}

  /**
   * Serialize an object to Json.
   *
   * @param object to serialize.
   */
  public String serialize(Object object, Class clazz) {
    return gson.toJson(object, clazz);
  }

  /**
   * Deserialize a json representation of an object.
   *
   * @param string A json string to deserialize.
   */
  public <T> T deserialize(String string, Class<T> clazz) {
    return gson.fromJson(string, clazz);
  }
}

File: kachelib/src/main/java/com/mskn73/kache/io/Serializer.kt
package com.mskn73.kache.io

import com.google.gson.Gson

class Serializer internal constructor() {

    private val gson = Gson()

    fun serialize(`object`: Any, clazz: Class<*>): String {
        return gson.toJson(`object`, clazz)
    }

    fun <T> deserialize(string: String, clazz: Class<T>): T {
        return gson.fromJson(string, clazz)
    }
}
